internal/fabric/schema/fabric_common_mapping_helpers.go

```diff
@@ -95,6 +95,9 @@ func LocationToFabric(locationList []interface{}) v4.SimplifiedLocation {
 }
 
 func LocationToTerra(location *v4.SimplifiedLocation) *schema.Set {
+	if location == nil {
```

Comment:
What was the issue with adding an empty list of location? Was it causing an error?

Reply:
I was getting nil pointer reference error because we didn't define metro code in the resource.

Reply:
metro_code is defined in the schema though. https://github.com/equinix/terraform-provider-equinix/blob/4d8095028106e24d178b4149eaa91c6d68b9316a/internal/fabric/schema/fabric_common_schemas.go#L138

How is it not defined in the resource? Help me understand this one.

Reply:
It's defined but I'm not passing it in my resource example block.

Reply:
This is from the API response though and not the input. Double check to see if location is being returned in the API response. If not, then we can leave this in. Please share log output.